File: dataprocessor.py
def process_string(input_text: str) -> str:
    if not input_text:
        return ""
        
    result = []
    count = 1
    
    for i in range(1, len(input_text)):
        if input_text[i] == input_text[i - 1]:
            count += 1
        else:
            result.append(f"{input_text[i - 1]}{count}")
            count = 1
            
    result.append(f"{input_text[-1]}{count}")
    
    return "".join(result)

def calculate_ratio(original: str, processed: str) -> float:
    if not original:
        return 0.0
    return len(processed) / len(original)

def main():
    test_data = "AAABBBCCDAA"
    processed_data = process_string(test_data)
    ratio = calculate_ratio(test_data, processed_data)
    
    print(f"Eredeti: {test_data}")
    print(f"Feldolgozott: {processed_data}")
    print(f"Arány: {ratio:.2f}")

if __name__ == "__main__":
    main()

File: Code/DataProcessor.java
import java.util.Scanner;

public class DataProcessor {
    
    public static String processString(String input) {
        if (input == null || input.isEmpty()) {
            return "";
        }
        
        StringBuilder result = new StringBuilder();
        int count = 1;
        
        for (int i = 1; i < input.length(); i++) {
            if (input.charAt(i) == input.charAt(i - 1)) {
                count++;
            } else {
                result.append(input.charAt(i - 1)).append(count);
                count = 1;
            }
        }
        result.append(input.charAt(input.length() - 1)).append(count);
        
        return result.toString();
    }

    public static double calculateRatio(String original, String processed) {
        if (original == null || original.isEmpty()) {
            return 0.0;
        }
        return (double) processed.length() / original.length();
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Kérlek, add meg a tömörítendő szöveget: ");
        String testData = scanner.nextLine();

        String processedData = processString(testData);
        double ratio = calculateRatio(testData, processedData);

        System.out.println("\n--- Eredmények ---");
        System.out.println("Eredeti adat: " + testData);
        System.out.println("Feldolgozott adat: " + processedData);
        System.out.println("Tömörítési arány: " + ratio);

        scanner.close();
    }
}